apps/sms/services/js/message_manager.js

```diff
@@ -421,33 +421,41 @@ var MessageManager = {
 
   markMessagesRead: function mm_markMessagesRead(list, isRead = true) {
     if (!this._mozMobileMessage || !list.length) {
-      return;
+      return Promise.reject(new Error('The mozMobileMessage API is not available or list is empty.'));
```

Comment:
I think we should return a resolved promise:
* either the list is empty, and it's OK
* or we don't have mozMobileMessage, and it usually means we're in Firefox. So we need to do as if everything works.